src/main/scala/viper/gobra/ast/internal/transform/CGEdgesTerminationTransform.scala

```diff
@@ -0,0 +1,156 @@
+      var definedFunctionsDelta: Map[in.FunctionProxy, in.FunctionLikeMember] = Map.empty
+
+      table.memberProxies.foreach {
+        case (t: in.InterfaceT, proxies) =>
+          val implementations = table.interfaceImplementations(t)
+          proxies.foreach {
+            case proxy: in.MethodProxy =>
+              table.lookup(proxy) match {
+                case m: in.Method if m.terminationMeasures.nonEmpty && m.body.isEmpty =>
+                  // only performs transformation if method has termination measures
+                  val src = m.getMeta
+                  val assumeFalse = in.Assume(in.ExprAssertion(in.BoolLit(b = false)(src))(src))(src)
+                  val newBody = {
+                    in.Block(
+                      decls = Vector.empty,
+                      stmts = assumeFalse +: implementations.toVector.flatMap { t: in.Type =>
+                        table.lookup(t, proxy.name).map {
+                          case implProxy: in.MethodProxy =>
+                            in.If(
+                              in.EqCmp(in.TypeOf(m.receiver)(src), typeAsExpr(t)(src))(src),
+                              in.Seqn(Vector(in.MethodCall(m.results map parameterAsLocalValVar, m.receiver, implProxy, m.args)(src), in.Return()(src)))(src),
+                              in.Seqn(Vector())(src)
+                            )(src)
+                          case v => Violation.violation(s"Expected a MethodProxy but got $v instead.")
+                        }
+                      }
+                    )(src)
+                  }
+                  val newMember = in.Method(m.receiver, m.name, m.args, m.results, m.pres, m.posts, m.terminationMeasures, Some(newBody))(src)
+                  methodsToRemove += m
+                  methodsToAdd += newMember
+                  definedMethodsDelta += proxy -> newMember
+
+                case m: in.PureMethod if m.terminationMeasures.nonEmpty && m.body.isEmpty =>
+                  // only performs transformation if method has termination measures
+                  val src = m.getMeta
+
+                  // the fallback function is called if no comparison succeeds
+                  val fallbackName = s"${m.name}$$fallback"
+                  val fallbackProxy = in.FunctionProxy(fallbackName)(src)
+                  val fallbackFunction = in.PureFunction(fallbackProxy, m.receiver +: m.args, m.results, m.pres, m.posts, m.terminationMeasures, None)(src)
+                  val returnType =
+                    if (m.results.length == 1) {
+                      m.results.head.typ
+                    } else if (m.results.length > 1) {
+                      in.TupleT(m.results.map(_.typ), Addressability.outParameter)
+                    } else {
+                      Violation.violation("Expected at least one out-parameter, but got 0 instead.")
+                    }
+                  val newBody = implementations.toVector.foldLeft[in.Expr](
```

Comment:
I thought the idea is to have the following body:
```
false ? /* do case split on dynamic type and perform calls to implementation */ : /* call fallback function */
```

Reply:
Adding to this, it would be great if you could express the transformations as comments (similarly to what we do in the encoding part)